### PR TITLE
feat: added edit note and change color of note

### PR DESCRIPTION
**Pull request by paarthbajaj**
edit

Known code improvements: remove commented code and console logs

### CODE REVIEW

File: src/components/NoteCard.css [hunk 1]
```diff
@@ -41,3 +42,65 @@
...
+  left: 50%;
+  transform: translate(-50%, -50%);
+  gap: 4px;
+  width: 19%;
+  flex-wrap: wrap;
+  padding: 8px 12px;
+  box-shadow: 0 1px 2px 0 rgb(60 64 67 / 30%), 0 2px 6px 2px rgb(60 64 67 / 15%);
+}
+.color-item {
+  height: 30px;
```
Comment (Kothalkarpriya) [inline src/components/NoteCard.css:76]: for more than 10 px you can use rem

File: src/components/NoteCard.css [hunk 2]
```diff
@@ -41,3 +42,65 @@
...
+  width: 30px;
+  border: 1px solid var(--grey-2);
+}
+.bgffffff {
+  background-color: #ffffff;
+}
```
Comment (Kothalkarpriya) [inline src/components/NoteCard.css:82]: colors you should use via variables everytime here

File: src/context/NotesContext.js [hunk 3]
```diff
@@ -6,18 +6,35 @@ import {
   useState,
 } from "react";
 import axios from "axios";
+const colorArr = [
+  "ffffff",
+  "f28b82",
+  "fbbc04",
+  "fff475",
+  "ccff90",
```
Comment (Kothalkarpriya) [inline src/context/NotesContext.js:14]: I liked this color array declarations, good idea

File: src/context/NotesContext.js [hunk 4]
```diff
@@ -54,7 +69,68 @@ const NotesContextProvider = ({ children }) => {
...
+    console.log(e.target.value);
+    setSelectedNote(() => ({ ...selectedNote, body: e.target.value }));
+  };
+  const updateNote = async (e) => {
+    e.preventDefault();
+    setCounter((counter) => counter + 1);
+    setIsPalleteOpen(() => false);
+    setIsModalOpen(() => false);
+    try {
+      const updatedNoteRes = await axios.post(
```
Comment (Kothalkarpriya) [inline src/context/NotesContext.js:92]: you used async-await that's good

File: src/context/NotesContext.js [hunk 5]
```diff
@@ -54,7 +69,68 @@ const NotesContextProvider = ({ children }) => {
...
+          },
+        }
+      );
+      console.log(updatedNoteRes);
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  const changeCardColor = async (e) => {
+    console.log(e);
```
Comment (Kothalkarpriya) [inline src/context/NotesContext.js:109]: if console output is not needed you can remove it as it is not necessary

**Review by Kothalkarpriya — APPROVED** [review]
Everything is good, just added some suggestions. Good work

File: src/context/NotesContext.js (hunk 5 above)
Comment (paarthbajaj) [reply src/context/NotesContext.js:109]: Yes I will remove it once I will complete P0 features.

File: src/components/NoteCard.css (hunk 2 above)
Comment (paarthbajaj) [reply src/components/NoteCard.css:82]: I am not using these colors again that's why have not used variables here.

File: src/components/AddNote.js [hunk 6]
```diff
@@ -53,10 +66,27 @@ export const AddNote = () => {
       <button onClick={saveClickHandler}>Save</button>
       <i className="fal fa-thumbtack position-absolute cursor-pointer" />
       <div className="btn-container position-absolute flex-row g-1 m-radius">
+        {/* <i
```
Comment (cjinc121) [inline src/components/AddNote.js:69]: Remove extra comments for cleaner code

File: src/components/NoteCard.css [hunk 7]
```diff
@@ -11,6 +11,7 @@
   padding: 0.5rem 0.5rem 2rem;
   box-shadow: 3px 3px 5px rgb(0 0 0 / 30%);
   width: 250px;
```
Comment (cjinc121) [inline src/components/NoteCard.css:13]: rem is preferable for size > 10px

File: src/components/NoteCard.css [hunk 8]
```diff
@@ -41,3 +42,65 @@
...
+  top: 0;
+  right: 0;
+  left: 0;
+  bottom: 0;
+  background-color: rgba(0, 0, 0, 0.5);
+  z-index: 10;
+}
+.edit-note-container {
+  background: var(--white);
+  /* position: absolute; */
```
Comment (cjinc121) [inline src/components/NoteCard.css:56]: again remove extra comments

**Review by cjinc121 — APPROVED** [review]
functionality working fine , some suggestions have been given for cleaner code

File: src/components/NoteCard.css (hunk 8 above)
Comment (paarthbajaj) [reply src/components/NoteCard.css:56]: Thank you for deep analysis, will remove once I complete P0 features.

File: src/components/NoteCard.css (hunk 2 above)
Comment (Kothalkarpriya) [reply src/components/NoteCard.css:82]: Okay

**Pull request merged**